confounds/metrics.py

```diff
@@ -6,3 +6,102 @@
 degree of harmonization achieved (e.g. reduction in variance of means/medians)
 
 """
+
+import numpy as np
+
+from confounds import Residualize
+from scipy import stats
+
+
+def partial_correlation(X, y=None):
+    """
+    Calculates the pairwise partial correlations between all of the variables in X with respect to some confounding variables y
```

Comment:
Hi James, it is best to rename the confound variable to C or something other than [small] y! You mention how its different from the typical y in the ML literature, it is easily missed by many. How about using C or conf_var or something like that?

Reply:
This one I wasn't sure even when writing what to use (I think I started with something like C) because partial correlation need not be with respect to confounds it could be any old variable. Happy to go with one of your suggestions as in the context of this package it is likely to always be confounds but just confirming that's what we want

Reply:
let's try use or make terminology that is likely to be used majority of our users, and also make a small note that they don't have to be confounds, as it is a generic computation.

that said, we are making a new function because it can not easily computed with scipy/numpy, yes?

Reply:
Have gone with C in the context of this package